cmd/juju/commands/switch_test.go

```diff
@@ -373,3 +366,130 @@ func (s *SwitchSimpleSuite) TestSwitchCurrentModelNoLongerInStore(c *gc.C) {
 	_, err := s.run(c, "mymodel")
 	c.Assert(err, gc.ErrorMatches, `"mymodel" is not the name of a model or controller`)
 }
+
+func (s *SwitchSimpleSuite) TestSwitchPreviousController(c *gc.C) {
+	s.store.CurrentControllerName = "currentCtrl"
+	s.store.PreviousControllerName = "previousCtrl"
+	s.addController(c, "currentCtrl")
+	s.addController(c, "previousCtrl")
+
+	// juju switch -: # Should switch to previous controller
+	context, err := s.run(c, "-:")
+
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(cmdtesting.Stderr(context), gc.Equals, "currentCtrl (controller) -> previousCtrl (controller)\n")
+}
+
+func (s *SwitchSimpleSuite) TestSwitchPreviousControllerWhichDoesntExists(c *gc.C) {
```

Comment:
Could you please add a test were you don't have any controller set in `PreviousControllerName`?